pkg/tester/data.go

```diff
@@ -152,17 +164,47 @@ func InitializeData(
 	}
 
 	opts := []database.BadgerOption{}
-	if config.CompressionDisabled {
-		opts = append(opts, database.WithoutCompression())
-	}
-	if config.MemoryLimitDisabled {
+	dataPathBackup := dataPath
+
+	if config.AllInMemoryEnabled{
 		opts = append(
 			opts,
-			database.WithCustomSettings(database.PerformanceBadgerOptions(dataPath)),
+			database.WithCustomSettings(database.AllInMemoryBadgerOptions(dataPath)),
+			database.WithoutCompression(),
 		)
+		dataPathBackup = ""
+	} else {
+		if config.CompressionDisabled {
+			opts = append(opts, database.WithoutCompression())
+		}
+		if config.MemoryLimitDisabled {
+			opts = append(
+				opts,
+				database.WithCustomSettings(database.PerformanceBadgerOptions(dataPath)),
+			)
+		}
+	}
+
+	if config.AllInMemoryEnabled || config.MemoryLimitDisabled {
```

Comment:
need comment to explain what does this section do

Reply:
updated